src/sql/finalize.sql

```diff
@@ -86,6 +86,11 @@ CREATE TYPE vector_accumulate_state (
     ALIGNMENT = double
 );
 
+CREATE TYPE svector_accumulate_state AS (
```

Comment:
I just use composite type rather than a custom type with in/out function here. It is not easy to reuse the state for sparse vector (length of `index` and `value` may change), so I think it is just okay to implement avg using plpgsql.

Reply:
We could just use an array length of which is `dims` since the result is probably dense.

Reply:
I am not sure about it. Do you think it is reasonable？ @gioeleperin

Reply:
> I am not sure about it. Do you think it is reasonable？ @gioeleperin

I'm not sure, as I don't have a deep understanding of how the underlying system works. However, I can say that when performing aggregations on sparse vectors, such as summing or averaging, the resulting vector is expected to remain sparse.

Reply:
If the the aggregation on sparse vectors is a sparse vector, we can perform a strategy that doubles the length of buffer if needed.

Reply:
it's still sparse. Can we specify buffer length here manually?

Reply:
> it's still sparse. Can we specify buffer length here manually?

The main problem is that we do know the length of the aggregation so we cannot allocate a buffer of proper size.

Reply:
> If the the aggregation on sparse vectors is a sparse vector, we can perform a strategy that we double the length of buffer if needed so that we can reuse the state for most iterations.

good idea.